Feature/smart field types

Obversity opened the pull request:
Improved the koi generators to include dragonfly accessors and some basic koi crud field types,

Obversity committed:
added smarter field types for images/files/enums in generators

Obversity committed:
added field config to all generator model templates, added permitted params for files

pvawser01 commented on lib/generators/koi.rb:
```diff
@@ -33,6 +33,98 @@ def process_attributes
+
+      #for keeping whitespace nice in form field type generation
+      def whitespace_for_field(attr)
+        "\s"*(@model_attributes.max_by{|attr|attr.name.chomp('_uid').length}.name.chomp('_uid').length-attr.name.chomp('_uid').length)
```
Break logic up on multiple lines for readability.

rahult commented:
:+1: awesome stuff

Obversity committed:
formatting fixes and making new generator code more readable

Obversity committed:
added belongs_to associations to smart field types in controller generators

Obversity committed:
fixes to formatting and guessing of field types / config in generators

Obversity committed:
updates to smarter generators
- accounted for polymorphic and indexed fields
- added some default validation for booleans
- refactored slightly

Obversity committed:
fixed attachment image helper so that it doesn't crash when provided the wrong file type

Obversity committed:
fixed file-type guessing method in generators to catch all _uid fields that the image guessing method doesn't

Obversity committed:
various fixes / improvements  to mailer / controller generators
